src/vector_of_array.jl

```diff
@@ -663,7 +663,20 @@ end
     bc = Broadcast.flatten(bc)
     N = narrays(bc)
     @inbounds for i in 1:N
-        if dest[:, i] isa AbstractArray && !isa(dest[:, i], StaticArraysCore.SArray)
+        if dest[:, i] isa AbstractArray && ArrayInterface.ismutable(dest[:, i])
+            copyto!(dest[:, i], unpack_voa(bc, i))
+        else
+            dest[:, i] = copy(unpack_voa(bc, i))
+        end
+    end
+    dest
+end
+
+@inline function Base.copyto!(dest::AbstractVectorOfArray,
+    bc::Broadcast.Broadcasted{<:Broadcast.DefaultArrayStyle})
+    bc = Broadcast.flatten(bc)
```

Comment:
won't this allocate?

Reply:
guess not, I'm adding a test for that.

Reply:
Which part? `unpack_voa` creates views, so it won't if the target array is mutable. If the target isn't, then it has to allocate